test/factories/workflow-specific/PIM_WorkflowFactory_v1.t.sol

```diff
@@ -360,45 +337,106 @@ contract PIM_WorkflowFactory_v1Test is E2ETest {
         vm.expectRevert(
             abi.encodeWithSelector(
                 Ownable.OwnableUnauthorizedAccount.selector, alice
             )
         );
-        factory.withdrawFee(token, alice);
+        factory.withdrawCreationFee(token, alice);
     }
+
+    function testWithdrawPimFee() public {
+        (IOrchestrator_v1 orchestrator, ERC20Issuance_v1 issuanceToken) =
+        factory.createPIMWorkflow(
+            workflowConfig,
+            paymentProcessorConfig,
+            logicModuleConfigs,
+            IPIM_WorkflowFactory_v1.PIMConfig({
+                fundingManagerMetadata: bancorVirtualSupplyBondingCurveFundingManagerMetadata,
+                authorizerMetadata: roleAuthorizerMetadata,
+                bcProperties: bcProperties,
+                issuanceTokenParams: issuanceTokenParams,
+                collateralToken: address(token),
+                recipient: alice,
+                isRenouncedIssuanceToken: true,
+                isRenouncedWorkflow: true
+            })
+        );
+
+        // CHECK: event is emitted from the bonding curve for fee withdrawal
+        // TODO: why does this fail if the fourth boolean is set to true?
+        vm.expectEmit(true, true, true, false);
+        emit IBondingCurveBase_v1.ProjectCollateralFeeWithdrawn(address(this), 0);
+        factory.withdrawPimFee(address(orchestrator.fundingManager()), alice);
+
+    }
+
+    // TODO: I dont understand why this test is failing. If I comment out the `vm.expectRevert` I can see that the `withdrawPimFee` function reverts
+
+    // function testWithdrawPimFee__FailsIfCallerIsNotPimFeeRecipient() public {
+    //     (IOrchestrator_v1 orchestrator, ERC20Issuance_v1 issuanceToken) =
+    //     factory.createPIMWorkflow(
+    //         workflowConfig,
+    //         paymentProcessorConfig,
+    //         logicModuleConfigs,
+    //         IPIM_WorkflowFactory_v1.PIMConfig({
+    //             fundingManagerMetadata: bancorVirtualSupplyBondingCurveFundingManagerMetadata,
+    //             authorizerMetadata: roleAuthorizerMetadata,
+    //             bcProperties: bcProperties,
+    //             issuanceTokenParams: issuanceTokenParams,
+    //             collateralToken: address(token),
+    //             recipient: workflowDeployer,
+    //             isRenouncedIssuanceToken: true,
+    //             isRenouncedWorkflow: true
+    //         })
+    //     );
+    //     // TODO: I also dont understand why using `vm.startPrank` behaves differently than using `vm.prank`
+
+    //     // 1. VERSION USING STARTPRANK
+    //     //
+    //     vm.startPrank(alice);
+    //     // vm.expectRevert();
+    //     factory.withdrawPimFee(address(orchestrator.fundingManager()), alice);
+    //     vm.stopPrank();
+
+    //     // 2. VERSION USING PRANK
+    //     // 
+    //     // // vm.expectRevert();
+    //     // vm.prank(alice);
+    //     // factory.withdrawPimFee(address(orchestrator.fundingManager()), alice);
```

Comment:
If I uncomment this instead, so that I have just the following lines, the call doesn't even revert anymore. It seems as if `prank` and `startPrank` are behaving differently in both cases.

```
        vm.prank(alice);
        factory.withdrawPimFee(address(orchestrator.fundingManager()), alice);
```